react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,103 @@
+import ReactDOM from 'react-dom';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {html: false, css: false, js: false};
+    this.jsOn = this.jsOn.bind(this);
+    this.jsRender = this.jsRender.bind(this);
+    this.cssOn = this.cssOn.bind(this);
+    this.htmlOn = this.htmlOn.bind(this);
+    this.htmlRender = this.htmlRender.bind(this);
+    this.cssRender = this.cssRender.bind(this);
+    this.start = this.start.bind(this);
+    this.startRender = this.startRender.bind(this);
+  }
+  jsOn() {
+    this.setState({ html: false, css: false, js: true});
+  }
+  cssOn(){
+    this.setState({html: false, css: true, js: false})
+  }
+  htmlOn(){
+    this.setState({html: true, css: false, js: false})
+  }
+  start() {
+    this.setState({ html: false, css: false, js: false })
+  }
+  htmlRender() {
+    return <div>
+            <div class="lang" onClick={this.start}>
+              <p>HyperText Markup Language</p>
+            </div>
+            <div class="text">
+              <p>{this.props.text[0].html}</p>
+            </div>
+            <div class="lang" onClick={this.cssOn}>
+              <p>Cascading Style Sheets</p>
+            </div>
+            <div class="lang" onClick={this.jsOn}>
+              <p>JavaScript</p>
+            </div>
+           </div>
+  }
+  cssRender() {
+    return <div>
+            <div class="lang" onClick={this.htmlOn}>
+              <p>HyperText Markup Language</p>
+            </div>
+            <div class="lang" onClick={this.start}>
+              <p>Cascading Style Sheets</p>
+            </div>
+            <div class="text">
+             <p>{this.props.text[0].css}</p>
+            </div>
+            <div class="lang" onClick={this.jsOn}>
+              <p>JavaScript</p>
+            </div>
+           </div>
+  }
+  jsRender() {
+    return <div>
+      <div class="lang" onClick={this.htmlOn}>
+        <p>HyperText Markup Language</p>
+      </div>
+      <div class="lang" onClick={this.cssOn}>
+        <p>Cascading Style Sheets</p>
+      </div>
+      <div class="lang" onClick={this.start}>
+        <p>JavaScript</p>
+      </div>
+      <div class="text">
+        <p>{this.props.text[0].js}</p>
+      </div>
+    </div>
+  }
+  startRender() {
+    return <div>
+            <div class="lang" onClick={this.htmlOn}>
+              <p>HyperText Markup Language</p>
+            </div>
+            <div class="lang" onClick={this.cssOn}>
+              <p>Cascading Style Sheets</p>
+            </div>
+            <div class="lang" onClick={this.jsOn}>
+              <p>JavaScript</p>
+            </div>
+            </div>
+  }
+  render() {
+    if (this.state.html === true) {
+      return this.htmlRender();
+    } else if (this.state.css === true) {
+      return this.cssRender();
+    } else if (this.state.js === true) {
+      return this.jsRender();
+    } else {
+      return this.startRender();
+    }
+  }
```

Comment:
You don't need any conditions here to determine what to render, your render method should look something like:

```JSX
render() {
  return (
    <div className="container">
      {this.props.items.map(item => {
        let contentClass = 'hide';
        
        if(this.state.id === item.id) {
          contentClass = 'show';
        }
        return (
          <div>
            <h4>{item.title}</h1>
            <p className={contentClass}>{item.content}</p>
          </div>
        )
      })}
    </div>
  );
  
  ```
  Notice we can use the id to compare with the id in state to determine which item should be open. This is not a full solution and probably doesn't match your variable and class names but it is to point you in the right direction.